pkg/controller/directimagemigration/directimagemigration_controller.go

```diff
@@ -71,8 +71,8 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	// Watch for changes to MigClusters referenced by DirectImageMigrations
 	err = c.Watch(
-		&source.Kind{Type: &migapi.MigCluster{}},
-		handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
+		source.Kind(mgr.GetCache(), &migapi.MigCluster{}),
```

Comment:
Please review this change, if you approve of this one, it's in a ton of places

Reply:
That is the proper way to do it